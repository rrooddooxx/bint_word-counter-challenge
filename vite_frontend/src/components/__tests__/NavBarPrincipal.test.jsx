import { render, screen } from "@testing-library/react";
import { NavBarPrincipal } from "../index.js";

describe("[COMP] <NavBarPrincipal>", () => {
  const navbarSections = [
    { name: "Inicio", href: "/" },
    { name: "API Docs", href: "#apidocs" },
  ];

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(
      <NavBarPrincipal navBarSections={navbarSections} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe renderizar el arreglo entregado por props como elementos del navbar", () => {
    const { container } = render(
      <NavBarPrincipal navBarSections={navbarSections} />
    );
    expect(screen.getAllByLabelText("navitem").length).toBe(
      navbarSections.length
    );
    expect(screen.getByText("Inicio").textContent).toBe(navbarSections[0].name);
    expect(screen.getByText("API Docs").textContent).toBe(
      navbarSections[1].name
    );
  });
});
