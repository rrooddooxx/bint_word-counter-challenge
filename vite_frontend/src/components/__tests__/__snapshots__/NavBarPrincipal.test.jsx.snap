// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMP] <NavBarPrincipal> Debe hacer match con el snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="#home"
      >
        ðŸš€ Word Ranking APP!
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            aria-label="navitem"
            class="nav-link"
            data-rr-ui-event-key="/"
            href="/"
          >
            Inicio
          </a>
          <a
            aria-label="navitem"
            class="nav-link"
            data-rr-ui-event-key="#apidocs"
            href="#apidocs"
          >
            API Docs
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
